src/h5web/visualizations/shared/utils.ts

```diff
@@ -45,37 +45,52 @@ export function computeVisSize(
     : { width, height: width / aspectRatio };
 }
 
+export function getExtent(
+  values: number[],
+  initialExtent?: [number, number, number]
+): [number, number, number] {
+  return values.reduce((acc, val) => {
+    const [accMin, accMax, accPositiveMin] = acc;
+    return [
+      Math.min(val, accMin),
+      Math.max(val, accMax),
+      val > 0 ? Math.min(val, accPositiveMin) : accPositiveMin,
+    ];
+  }, initialExtent || [values[0], values[0], Infinity]);
+}
+
 export function getDomain(
   values: number[],
-  scaleType: ScaleType = ScaleType.Linear
+  scaleType: ScaleType = ScaleType.Linear,
+  errors?: number[]
 ): Domain | undefined {
   if (values.length === 0) {
     return undefined;
   }
 
-  const [min, max, positiveMin] = values.reduce<
-    [number, number, number | undefined]
-  >(
-    (acc, val) => {
-      const [accMin, accMax, accPositiveMin] = acc;
-      return [
-        accMin > val ? val : accMin,
-        accMax < val ? val : accMax,
-        val > 0 && (accPositiveMin === undefined || accPositiveMin > val)
-          ? val
-          : accPositiveMin,
-      ];
-    },
-    [values[0], values[0], undefined]
-  );
+  if (errors && errors.length !== values.length) {
+    throw new Error(
+      'Errors when computing domain should have the same size as values'
```

Comment:
I would word this `Expected 'errors' array to have same length as 'values' array` for consistency.

`when computing domain` isn't very useful to the user, and for us, we can easily find the error in the code (or in the stack trace).